### PR TITLE
Speed up backend tests

### PR DESCRIPTION
**Pull request by taehoonlee**
### Summary

This PR speeds up backend tests by removing redundant backend tests. The targets, which always been included in the slowest 20 test durations, are `test_elementwise_operations` and `test_nn_operations`.

This PR also cleans up old styles for backend tests including `assert_value_with_ref`, `return_results`, `shape_or_val`, and the related stuff with them. As a result, the `test_conv`, `test_depthwise_conv`, and `test_pool` are now much cleaner than before.

### Related Issues

#10972
#10956
#10930

cc @gabrieldemarmiesse

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
What does cc mean? Should I review this PR?

**Comment by taehoonlee** [comment]
@gabrieldemarmiesse I mean that please feel free to see the changes :) Because you have been contributed to related issues.

**Comment by gabrieldemarmiesse** [comment]
Sure then, thanks!

File: tests/keras/backend/backend_test.py [hunk 1]
```diff
@@ -36,6 +35,9 @@
     warnings.warn('Could not import the Theano backend')
 
 
+WITH_NP = [KTH if K.backend() == 'theano' else KC if K.backend() == 'cntk' else KTF, KNP]
```
Comment (gabrieldemarmiesse) [inline tests/keras/backend/backend_test.py]: I must be missing something. Why this rather than `WITH_NP = [K, KNP]`?

Comment (gabrieldemarmiesse) [reply tests/keras/backend/backend_test.py]: I see, it's to be able to compare it with the other backends with `==`. Nevermind.

**Comment by gabrieldemarmiesse** [comment]
From what I see, it looks nice and clean now. There doesn't seem to be any weird stuff going on.

**Pull request merged**